src/main/target/common_pre.h

```diff
@@ -74,6 +74,7 @@
 
 #ifdef STM32F7
 #define USE_ITCM_RAM
+#define ITCM_RAM_OPTIMISATION "-O2"
```

Comment:
I can confirm that this does indeed reduce code side, and also that when code is built in a debug environment, the code seems to be debuggable. However the documentation states

> `optimize` function attribute The `optimize` attribute is used to specify that a function is to be compiled with different optimization options than specified on the command line.

It would therefore be cleaner, if when -DDEBUG is passed on the command line that this is picked up here and ITCM_RAM_OPTIMISATION only defined when not performing a debug build as under those circumstances we want `-Og` to take precedence.

Reply:
@SteveCEvans: Interesting point about debugging - I can add this to this pull request.

Reply:
@SteveCEvans: Amended - now the optimisation level for debug builds will be respected for ITCM_RAM.